### PR TITLE
Fixes selection of kubeadm-criconfig version

### PR DESCRIPTION
**Pull request by chentex**
Fixes selection of `kubeadm-criconfig` version.

When removing the `kubeadm-criconfig` package zypper will install freely any version it finds more suitable, if there is a new one it will install that one.

## Why is this PR needed?
We also need to install cri-o-1.18-kubeadm-criconfig at the same time that we remove cri-o-kubeadm-criconfig, because otherwise the kubernetes-1.18-kubeadm requirements would trigger the installation of any cri-o-*-kubeadm-criconfig leading to unexpected results on late migrations to SP2 and therefore late upgrades to caasp 4.5
Does it fix an issue? addresses a business case?

Relates https://github.com/SUSE/avant-garde/issues/1756

**Reminder**: Add the "fixes bsc#XXXX" to the title of the commit so that it will
appear in the changelog.

## What does this PR do?

Installs cri-o-1.18-kubeadm-criconfig at the same time that we remove cri-o-kubeadm-criconfig

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This needs to be tested against the test repository

https://build.suse.de/project/show/Devel:CaaSP:5:Branches:PackageUpgradeTesting

- Deploy a v4 cluster in version 1.17.4
- Perform an upgrade to 1.18.5

All should work.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -398,6 +398,7 @@ github.com/pkg/errors v0.8.0/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINE
 github.com/pkg/errors v0.8.1/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
 github.com/pkg/errors v0.9.1 h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=
 github.com/pkg/errors v0.9.1/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
+github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
```
Comment (evrardjp) [inline go.sum:401]: That seems outside the scope of this PR :)

Comment (chentex) [reply go.sum:401]: Will remove it, it was done automatically.

Comment (chentex) [reply go.sum]: done
